ddmrp/models/stock_buffer.py

```diff
@@ -1236,23 +1242,29 @@ def cron_actions(self, only_nfp=False):
     def cron_ddmrp(self, automatic=False):
         """Calculate key DDMRP parameters for each buffer.
         Called by cronjob."""
+        auto_commit = not getattr(threading.currentThread(), "testing", False)
         _logger.info("Start cron_ddmrp.")
-        buffers = self.search([])
+        buffer_ids = self.search([]).ids
+
         i = 0
-        j = len(buffers)
-        buffers.refresh()
-        for b in buffers:
-            i += 1
-            _logger.debug("ddmrp cron: {}. ({}/{})".format(b.name, i, j))
-            try:
-                if automatic:
-                    with self.env.cr.savepoint():
+        j = len(buffer_ids)
+        for buffer_chunk_ids in split_every(50, buffer_ids):
+            for b in self.browse(buffer_chunk_ids):
+                i += 1
+                _logger.debug("ddmrp cron: {}. ({}/{})".format(b.name, i, j))
+                try:
+                    if automatic:
+                        with self.env.cr.savepoint():
+                            b.refresh()
```

Comment:
any idea why the cache has to be invalidated on each record? (btw `refresh` is deprecated since 8.0 :wink:

Reply:
@JordiBForgeFlow when doing so, all the ids in buffer_chunk_ids are re-read from database on each loop, so if it's really required, then it's better at least to have the `browse` inside the loop... but that'd be a pity :(

Reply:
Moreover, this PR adds an `invalidate_cache` in `cron_actions` as well: https://github.com/OCA/ddmrp/pull/59

Reply:
And same remark as https://github.com/OCA/ddmrp/pull/59#discussion_r568584991, the invalidation of cache should be restricted to some ids and only for the computed fields that needs it, so we don't need to re-read all the buffers on each loop